install/Knative-with-Minishift.md

```diff
@@ -0,0 +1,256 @@
+- The  [addon](https://docs.okd.io/latest/minishift/using/addons.html) **anyuid** allows the `default` service account to run the application with uid `0`
+
+- The command `minishift profile set knative` is required every time you start and stop minishift to make sure that you are on right `knative` minishift profile that was configured above.
+
+## Configuring `oc` (openshift cli)
+
+Running the following command make sure that you have right version of `oc` and have configured the DOCKER daemon to be connected to minishift docker.
+
+```shell
+# configures the host talk to DOCKER daemon of minishift
+minishift docker-env
+# configures the right openshift cli
+minishift oc-env
+```
+
+## Preparing Knative Deployment
+
+### Enable Admission Controller Webhook
+To be able to deploy and run serverless Knative applications, its required that you must enable the [Admission Controller Webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).  
+
+Run the following command to make OpenShift (run via minishift) to be configured for [Admission Controller Webhook](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/):
+
+```shell
+# Enable admission controller webhooks
+# The configuration stanzas below look weird and are just to workaround for:
+# https://bugzilla.redhat.com/show_bug.cgi?id=1635918
+minishift openshift config set --target=kube --patch '{
+    "admissionConfig": {
+        "pluginConfig": {
+            "ValidatingAdmissionWebhook": {
+                "configuration": {
+                    "apiVersion": "apiserver.config.k8s.io/v1alpha1",
+                    "kind": "WebhookAdmission",
+                    "kubeConfigFile": "/dev/null"
+                }
+            },
+            "MutatingAdmissionWebhook": {
+                "configuration": {
+                    "apiVersion": "apiserver.config.k8s.io/v1alpha1",
+                    "kind": "WebhookAdmission",
+                    "kubeConfigFile": "/dev/null"
+                }
+            }
+        }
+    }
+}'
+
+# wait until the kube-apiserver is restarted
+until oc login -u admin -p admin; do sleep 5; done;
+```
+
+### Configuring a OpenShift project
+
+1. Set up the project **myproject** for use with Knative applications.
+
+    ```shell
+    oc project myproject
+    oc adm policy add-scc-to-user privileged -z default
+    oc label namespace myproject istio-injection=enabled
+    ```
+    The `oc adm policy` adds the **privileged** [Security Context Constraints(SCCs)](https://docs.okd.io/3.10/admin_guide/manage_scc.html) to the **default** Service Account. The SCCs are the precursor to the PSP (Pod Security Policy) mechanism in Kubernetes.
```

Comment:
Could be worthwhile adding *why* we're doing this (Because istio sidecars need it)